File: backend/src/main/java/com/bjtufood/canteen/entity/Canteen.java
package com.bjtufood.canteen.entity;

import com.baomidou.mybatisplus.annotation.*;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 食堂实体类
 * <p>
 * 对应数据库表：canteen
 * 例如：第一食堂、第二食堂
 */
@Data
@TableName("canteen")
@Schema(description = "食堂")
public class Canteen {

    @TableId(type = IdType.AUTO)
    @Schema(description = "食堂ID")
    private Long id;

    /** 食堂名称 */
    @Schema(description = "食堂名称", example = "第一食堂")
    private String name;

    /** 食堂图片 URL 列表，JSON 字符串 */
    @Schema(description = "食堂图片URL列表JSON")
    private String images;

    /** 食堂位置 */
    @Schema(description = "食堂位置")
    private String location;

    /** 纬度（GCJ-02，距离排序用） */
    @Schema(description = "纬度（GCJ-02）")
    private BigDecimal latitude;

    /** 经度（GCJ-02，距离排序用） */
    @Schema(description = "经度（GCJ-02）")
    private BigDecimal longitude;

    /** 食堂描述 */
    @Schema(description = "食堂描述")
    private String description;

    /** 状态：open / closed */
    @Schema(description = "状态", example = "open")
    private String status;

    /**
     * 审核状态（与启停 status 解耦）：pending（待审核）/ approved（已通过）/ rejected（已退回）
     * 后台录入默认 approved；学生 UGC 提交写入 pending。
     */
    @Schema(description = "审核状态：pending/approved/rejected", example = "approved")
    private String auditStatus;

    /** 退回原因（仅 audit_status=rejected 时由后台填写，可空） */
    @Schema(description = "退回原因（audit_status=rejected 时由后台填写）")
    private String rejectReason;

    /** 提交人用户ID（UGC 由当前登录用户写入，禁止前端传入） */
    @Schema(description = "提交人用户ID")
    private Long createdBy;

    /** 排序权重（数字越小越靠前） */
    @Schema(description = "排序权重")
    private Integer sortOrder;

    @TableField(fill = FieldFill.INSERT)
    @Schema(description = "创建时间")
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    @Schema(description = "更新时间")
    private LocalDateTime updatedAt;
}

File: frontend/src/types/canteen.ts
/** 食堂信息 */
export interface CanteenInfo {
  name: string
  location: string
  icon: string
  /** 与用户位置的距离（米，按定位排序时后端返回）；未定位为 undefined */
  distance?: number
}

/** 档口详情 */
export interface StallDetail {
  id?: number
  name: string
  images: string[]
  location: string
  description: string
  /** 平均星级（后端 StallDetailVO.avgRating，BigDecimal，无则 0.00） */
  avgRating?: number
  /** 标徽/标签（如 招牌/清真；后端暂未返回，前端可从菜品 tags 派生，见 stall.vue stallTags） */
  tags?: string[]
}

/** 档口简讯（GET /stalls 列表项） */
export interface StallInfo {
  id: number
  canteenId?: number
  name: string
  images?: string[]
  location?: string
  description?: string
  avgRating?: number
  status?: 'open' | 'closed'
}

/** 我的档口·食堂 UGC 提交（GET /my/stalls，对齐 StallAdminVO：auditStatus / rejectReason） */
export interface MyPublishStall {
  id: number
  /** 提交类型：stall=档口，canteen=食堂 */
  type: 'stall' | 'canteen'
  name: string
  location?: string
  description?: string
  images?: string[]
  auditStatus: 'pending' | 'approved' | 'rejected'
  rejectReason?: string
  createTime?: string
}
